types/span.go

```diff
@@ -35,31 +34,24 @@ type CoreSpanMetadata struct {
 	DurationMs   float64 `json:"durationMs,omitempty"`
 }
 
-// convertTimestamp turns a Zipkin timestamp (a Unix timestamp in microseconds)
+type SpanAnnotation struct {
+	Timestamp int64         `json:"timestamp"`
+	Value     string        `json:"value"`
+	Host      *SpanEndpoint `json:"endpoint,omitempty"`
+}
+
+type SpanEndpoint struct {
+	Ipv4        string `json:"ipv4"`
+	Port        int    `json:"port"`
+	ServiceName string `json:"serviceName"`
+}
+
+// ConvertTimestamp turns a Zipkin timestamp (a Unix timestamp in microseconds)
 // into a time.Time value.
-func convertTimestamp(tsMicros int64) time.Time {
+func ConvertTimestamp(tsMicros int64) time.Time {
 	if tsMicros == 0 {
 		return time.Now().UTC()
 	}
 
 	return time.Unix(tsMicros/1000000, (tsMicros%1000000)*1000).UTC()
 }
-
-// guessAnnotationType takes a string value and turns it into a bool, int64 or
-// float64 value if possible. This is a workaround for the fact that Zipkin
-// BinaryAnnotation values are always transmitted as strings.
-// (See e.g. the Zipkin API spec here:
-// https://github.com/openzipkin/zipkin-api/blob/72280f3/zipkin-api.yaml#L235-L245)
-func guessAnnotationType(v string) interface{} {
```

Comment:
Moved to v1/json.go, only place where this is used

This is part of another PR of mine: https://github.com/honeycombio/honeycomb-opentracing-proxy/pulls